src/main/java/com/google/sps/servlets/FreeTimeServlet.java

```diff
@@ -0,0 +1,211 @@
+    calendarClientFactory = new CalendarClientImpl.Factory();
+    gmailClientFactory = new GmailClientImpl.Factory();
+  }
+
+  /**
+   * Create servlet with explicit implementations of CalendarClient and AuthenticationVerifier
+   *
+   * @param authenticationVerifier implementation of AuthenticationVerifier
+   * @param calendarClientFactory implementation of CalendarClientFactory
+   */
+  public FreeTimeServlet(
+      AuthenticationVerifier authenticationVerifier,
+      CalendarClientFactory calendarClientFactory,
+      GmailClientFactory gmailClientFactory) {
+    super(authenticationVerifier);
+    this.calendarClientFactory = calendarClientFactory;
+    this.gmailClientFactory = gmailClientFactory;
+  }
+
+  /**
+   * Returns string containing the some of the user's free time intervals, where events can be
+   * created
+   *
+   * @param request Http request from the client. Should contain idToken and accessToken
+   * @param response 403 if user is not authenticated, or Json string with the user's events
+   * @throws IOException if an issue arises while processing the request
+   */
+  @Override
+  public void doGet(
+      HttpServletRequest request, HttpServletResponse response, Credential googleCredential)
+      throws IOException {
+    assert googleCredential != null
+        : "Null credentials (i.e. unauthenticated requests) should already be handled";
+
+    CalendarClient calendarClient = calendarClientFactory.getCalendarClient(googleCredential);
+    long fiveDaysInMillis = TimeUnit.DAYS.toMillis(5);
+    Date timeMin = calendarClient.getCurrentTime();
+    Date timeMax = Date.from(timeMin.toInstant().plus(Duration.ofDays(5)));
+    List<Event> calendarEvents = getEvents(calendarClient, timeMin, timeMax);
+    // Initialize the freeTime utility. Keep track of the free time in the next 5 days, with
+    // work hours as defined between 10am and 6 pm. The rest of the time between 7 am and 11 pm
+    // should be considered personal time.
+    int personalBeginHour = 7;
+    int workBeginHour = 10;
+    int workEndHour = 18;
+    int personalEndHour = 23;
+    int numDays = 5;
+    FreeTimeUtility freeTimeUtility =
+        new FreeTimeUtility(
+            timeMin, personalBeginHour, workBeginHour, workEndHour, personalEndHour, numDays);
+    for (Event event : calendarEvents) {
+      DateTime start = event.getStart().getDateTime();
+      start = start == null ? event.getStart().getDate() : start;
+      DateTime end = event.getEnd().getDateTime();
+      end = end == null ? event.getEnd().getDate() : end;
+      Date eventStart = new Date(start.getValue());
+      Date eventEnd = new Date(end.getValue());
+      if (eventStart.before(timeMin)) {
+        eventStart = timeMin;
+      }
+      if (eventEnd.after(timeMax)) {
+        eventEnd = timeMax;
+      }
+      freeTimeUtility.addEvent(eventStart, eventEnd);
+    }
+
+    int wordCount = getWordCount(googleCredential);
+    int averageReadingSpeed = 50;
```

Comment:
Also make this a class constant for now.

Reply:
`Resolved`

Reply:
`Resolved`